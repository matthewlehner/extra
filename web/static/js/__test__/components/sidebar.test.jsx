import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Sidebar from "components/sidebar";

describe("Sidebar component", () => {
  it("renders correctly", () => {
    const data = {
      loading: false,
      channels: [{ id: "1", name: "something else", provider: "twitter" }],
      collections: [{ id: "1", name: "something" }]
    };
    const tree = renderer.create(
      <MemoryRouter>
        <Sidebar data={data} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders loading state", () => {
    const data = {
      loading: true,
      channels: [{ id: "1", name: "something else", provider: "twitter" }],
      collections: [{ id: "1", name: "something" }]
    };
    const tree = renderer.create(
      <MemoryRouter>
        <Sidebar data={data} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders an error state", () => {
    const data = {
      error: true,
      loading: false,
      channels: [{ id: "1", name: "something else", provider: "twitter" }],
      collections: [{ id: "1", name: "something" }]
    };
    const tree = renderer.create(
      <MemoryRouter>
        <Sidebar data={data} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
