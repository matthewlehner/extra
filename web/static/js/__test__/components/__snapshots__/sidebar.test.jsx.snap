// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders an error state 1`] = `<div />`;

exports[`Sidebar component renders correctly 1`] = `
<section
  className="sidebar"
>
  <svg
    className="extra-logo"
  >
    <use
      xlinkHref="#id"
    />
  </svg>
  <nav
    className="site-nav"
  >
    <section>
      <h3>
        Collections
      </h3>
      <ul>
        <li>
          <a
            className={undefined}
            href="/collections/1"
            onClick={[Function]}
            style={undefined}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            something
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Add new
          </a>
        </li>
      </ul>
    </section>
    <section>
      <h3>
        Channels
      </h3>
      <ul>
        <li>
          <a
            className={undefined}
            href="/channels/1"
            onClick={[Function]}
            style={undefined}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            something else
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Add new
          </a>
        </li>
      </ul>
    </section>
    <section>
      <h3>
        General
      </h3>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Performance
          </a>
        </li>
        <li>
          <a
            href="/account"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Account
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Logout
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="gradient-container"
  >
    <div
      className="gradient"
    />
  </div>
</section>
`;

exports[`Sidebar component renders loading state 1`] = `
<section
  className="sidebar"
>
  <svg
    className="extra-logo"
  >
    <use
      xlinkHref="#id"
    />
  </svg>
  <nav
    className="site-nav"
  >
    <section>
      <h3>
        Collections
      </h3>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Add new
          </a>
        </li>
      </ul>
    </section>
    <section>
      <h3>
        Channels
      </h3>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Add new
          </a>
        </li>
      </ul>
    </section>
    <section>
      <h3>
        General
      </h3>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Performance
          </a>
        </li>
        <li>
          <a
            href="/account"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Account
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="#id"
              />
            </svg>
            Logout
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="gradient-container"
  >
    <div
      className="gradient"
    />
  </div>
</section>
`;
